backend/compact-connect/tests/unit/test_app.py

```diff
@@ -205,16 +225,40 @@ def _inspect_persistent_stack(
             local_ui_port = '3018' if not local_ui_port else local_ui_port
             callbacks.append(f'http://localhost:{local_ui_port}/auth/callback')
 
-        # Ensure our Staff user pool is configured with the expected callbacks
+        # ensure we have two user pools, one for staff users and one for providers
+        persistent_stack_template.resource_count_is(CfnUserPool.CFN_RESOURCE_TYPE_NAME, 2)
+
+        # Ensure our Staff user pool is configured with the expected callbacks and read/write attributes
         persistent_stack_template.has_resource(
             type=CfnUserPoolClient.CFN_RESOURCE_TYPE_NAME,
             props={
                 'Properties': {
-                    'CallbackURLs': callbacks
+                    'CallbackURLs': callbacks,
+                    "ReadAttributes": ["email"],
+                    "WriteAttributes": ["email"],
                 }
             }
         )
 
+        # Ensure our Provider user pool is created with expected values
+        persistent_stack_template.has_resource_properties(
```

Comment:
To be sure you're inspecting the UserPool you expect to be, I think you could use the `stack.get_logical_id(CfnElement)` in combination with `dict` returned by `template.find_resources(CfnUserPoolClient.CFN_RESOURCE_TYPE_NAME)` to look up exactly the template element you want and inspect it directly.